contracts/service-registry/src/contract/execute.rs

```diff
@@ -84,25 +85,30 @@ pub fn bond_verifier(
         .may_load(deps.storage, &service_name)?
         .ok_or(ContractError::ServiceNotFound)?;
 
-    let bond = if !info.funds.is_empty() {
-        info.funds
-            .iter()
-            .find(|coin| coin.denom == service.bond_denom)
-            .ok_or(ContractError::WrongDenom)?
-            .amount
+    let bond: Option<nonempty::Uint128> = if !info.funds.is_empty() {
+        Some(
+            info.funds
+                .iter()
+                .find(|coin| coin.denom == service.bond_denom)
+                .ok_or(ContractError::NoFundsToBond)?
```

Comment:
this is misleading. This will also trigger if someone sends the wrong denomination